app/controllers/smart_proxies_controller.rb

```diff
@@ -41,12 +46,21 @@ def refresh
   end
 
   def version
-    begin
-      version = @smart_proxy.version
-    rescue Foreman::Exception => exception
-      render :json => {:success => false, :message => exception.message} and return
+    Rails.cache.fetch("proxy_#{@smart_proxy.id}/version", :expires_in => 3.minutes) do
+      requested_data(:version)
     end
-    render :json => {:success => true, :message => version[:message]}
+  end
+
+  def plugin_version
+    plugin_name = params[:plugin]
+    render :json => {:success => false, :message => _('Plugin name cannot be blank')} and return if plugin_name.blank?
+    Rails.cache.fetch("proxy_#{@smart_proxy.id}/#{plugin_name}/version", :expires_in => 3.minutes) do
```

Comment:
does it make sense to cache the value per proxy version? AFAIR you get it all in one hash, so maybe cache the entire hash?